src/main/java/edu/harvard/iq/dataverse/dataaccess/StorageIO.java

```diff
@@ -606,7 +606,8 @@ public static String getDriverPrefix(String driverId) {
     }
     
     public static boolean isDirectUploadEnabled(String driverId) {
-        return Boolean.parseBoolean(System.getProperty("dataverse.files." + driverId + ".upload-redirect"));
+        return Boolean.parseBoolean(System.getProperty("dataverse.files." + driverId + ".upload-redirect")) ||
```

Comment:
Since direct-upload only works with S3 stores, should the base class just check out-of-band and the S3 store override this method to check both?

Reply:
This is a little bit confusing. It is a static method. Refactoring the code does not seem to be the best solution to me. For now, I have changed the code to this (which seems equivalent to me):
```
    public static boolean isDirectUploadEnabled(String driverId) {
        return (DataAccess.S3.equals(driverId) && Boolean.parseBoolean(System.getProperty("dataverse.files." + DataAccess.S3 + ".upload-redirect"))) ||
            Boolean.parseBoolean(System.getProperty("dataverse.files." + driverId + ".upload-out-of-band"));
    }
```